Tests/SwiftParserTest/IncrementalParsingTests.swift

```diff
@@ -61,3 +115,46 @@ public class IncrementalParsingTests: XCTestCase {
     XCTAssertEqual(origStructB.parent!.id, reusedNode.parent!.id)
   }
 }
+
+// MARK: helper functions
+extension IncrementalParsingTests {
+  /// return the diff of two strings with SourceEdit
+  /// example:
+  ///  - s1: 0123456
+  ///  - s2: x12456yz
+  ///  should return [(0, 1, 0,), (0, 0, 1), (3, 1, 0), (6, 0, 1), (7, 0, 1)]
+  ///  the above edits are **serial**
```

Comment:
Instead of explicitly specifying that these edits are serial, what do you think about returning `ConcurrentEdits` here? That way the Swift type system will make sure we aren’t mixing up sequential and concurrent edits.

Reply:
Since our goal is to use this function to provide `ConcurrentEdits` for `IncrementalParseTransition`, that makes sense to me.👍